src/tools/cephfs_mirror/main.cc

```diff
@@ -47,12 +48,32 @@ int main(int argc, const char **argv) {
                          CODE_ENVIRONMENT_DAEMON,
                          CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS);
 
-  if (g_conf()->daemonize) {
-    global_init_daemonize(g_ceph_context);
```

Comment:
So, this was causing valgrind to run report `Leak_StillReachable` kind of errors when `cephfs-mirror` is run without -f/-d (no `Leak_StillReachable` errors when run with -f with the PR minus this commit).

```
<error>
  <unique>0x0</unique>
  <tid>1</tid>
  <kind>Leak_StillReachable</kind>
  <xwhat>
    <text>4 bytes in 1 blocks are still reachable in loss record 1 of 401</text>
    <leakedbytes>4</leakedbytes>
    <leakedblocks>1</leakedblocks>
  </xwhat>
  <stack>
    <frame>
      <ip>0x4835DEF</ip>
      <obj>/usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so</obj>
      <fn>operator new(unsigned long)</fn>
    </frame>
    <frame>
      <ip>0x675DB93</ip>
      <obj>/work/ceph/build/lib/libceph-common.so.2</obj>
      <fn>ceph::common::CephContext::CephContext(unsigned int, code_environment_t, int)</fn>
      <dir>/work/ceph/src/common</dir>
      <file>ceph_context.cc</file>
      <line>734</line>
    </frame>
    <frame>
      <ip>0x67D950D</ip>
      <obj>/work/ceph/build/lib/libceph-common.so.2</obj>
      <fn>common_preinit(CephInitParameters const&amp;, code_environment_t, int)</fn>
      <dir>/work/ceph/src/common</dir>
      <file>common_init.cc</file>
      <line>37</line>
    </frame>
    <frame>
      <ip>0x279AD9</ip>
      <obj>/work/ceph/build/bin/cephfs-mirror</obj>
      <fn>global_pre_init(std::map&lt;std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt;, std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt;, std::less&lt;std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; &gt;, std::allocator&lt;std::pair&lt;std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; const, std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; &gt; &gt; &gt; const*, std::vector&lt;char const*, std::allocator&lt;char const*&gt; &gt;&amp;, unsigned int, code_environment_t, int)</fn>
      <dir>/work/ceph/src/global</dir>
      <file>global_init.cc</file>
      <line>119</line>
    </frame>
    <frame>
      <ip>0x27A096</ip>
      <obj>/work/ceph/build/bin/cephfs-mirror</obj>
      <fn>global_init(std::map&lt;std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt;, std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt;, std::less&lt;std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; &gt;, std::allocator&lt;std::pair&lt;std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; const, std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; &gt; &gt; &gt; const*, std::vector&lt;char const*, std::allocator&lt;char const*&gt; &gt;&amp;, unsigned int, code_environment_t, int, bool)</fn>
      <dir>/work/ceph/src/global</dir>
      <file>global_init.cc</file>
      <line>195</line>
    </frame>
    <frame>
      <ip>0x1FFCF0</ip>
      <obj>/work/ceph/build/bin/cephfs-mirror</obj>
      <fn>main</fn>
      <dir>/work/ceph/src/tools/cephfs_mirror</dir>
      <file>main.cc</file>
      <line>48</line>
    </frame>
  </stack>
</error>
```

I changed it to how it is done in ceph_osd.cc (and other daemons) and valgrind is happy. Is calling `global_init_daemonize` incorrect w.r.t. running with valgrind? @tchaikov any idea?